src/test/java/io/specto/hoverfly/junit/core/HoverflyTest.java

```diff
@@ -566,20 +570,20 @@ public void shouldSetModeArgumentsForCaptureMode() {
     @Test
     public void shouldResetModeWithModeArguments() {
         hoverfly = new Hoverfly(localConfigs().captureAllHeaders().enableStatefulCapture(), CAPTURE);
-
         HoverflyClient hoverflyClient = createMockHoverflyClient(hoverfly);
+        when(hoverflyClient.getHealth()).thenReturn(true);
```

Comment:
This test becomes harder to understand and testing both `hoverfly.start()` and `hoverfly.resetMode()` all together. We should just mock HoverflyClient only to keep test simple and less coupled to implementation details: 
```
Whitebox.setInternalState(hoverfly, "hoverflyClient", hoverflyClient);
```